src/array_value.cpp

```diff
@@ -507,4 +526,43 @@ ValuePtr ArrayValue::uniq(Env *env) {
     return hash->keys(env);
 }
 
+ValuePtr ArrayValue::clear(Env *env) {
+    this->assert_not_frozen(env);
```

Comment:
Noticed `assert_not_frozen` in pop when should this be called?
Anytime when changing the underlying value?

Reply:
Yes, any time we are mutating the underlying object rather than returning a copy, we should add `assert_not_frozen`.